standard/src/main/java/net/kyori/moonshine/strategy/StandardPlaceholderResolverStrategy.java

```diff
@@ -0,0 +1,168 @@
+//
+// moonshine - A localisation library for Java.
+// Copyright (C) 2021 Mariell Hoversholm
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+package net.kyori.moonshine.strategy;
+
+import com.google.common.collect.Iterators;
+import io.leangen.geantyref.GenericTypeReflector;
+import java.lang.reflect.Parameter;
+import java.lang.reflect.Type;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.NavigableSet;
+import javax.annotation.concurrent.ThreadSafe;
+import net.kyori.moonshine.Moonshine;
+import net.kyori.moonshine.annotation.Placeholder;
+import net.kyori.moonshine.exception.PlaceholderResolvingException;
+import net.kyori.moonshine.exception.UnfinishedPlaceholderException;
+import net.kyori.moonshine.model.MoonshineMethod;
+import net.kyori.moonshine.placeholder.ConclusionValue;
+import net.kyori.moonshine.placeholder.ContinuanceValue;
+import net.kyori.moonshine.placeholder.IPlaceholderResolver;
+import net.kyori.moonshine.strategy.supertype.ISupertypeStrategy;
+import net.kyori.moonshine.util.Either;
+import net.kyori.moonshine.util.Weighted;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+@ThreadSafe
+public final class StandardPlaceholderResolverStrategy<R, I, F> implements
```

Comment:
This could really do with explanation - at a glance I could only assume that this this implementation was the "right" one to use based on it being used in the tests, and I still don't understand what it does.